doc/manual/src/glossary.md

```diff
@@ -1,11 +1,95 @@
 # Glossary
 
   - [derivation]{#gloss-derivation}\
-    A description of a build task. The result of a derivation is a
-    store object. Derivations are typically specified in Nix expressions
-    using the [`derivation` primitive](language/derivations.md). These are
-    translated into low-level *store derivations* (implicitly by
-    `nix-env` and `nix-build`, or explicitly by `nix-instantiate`).
+    A [build] task declared in the [Nix language](./language/index.md) using the built-in function [`derivation`](language/derivations.md).
+
+    [derivation]: #gloss-derivation
+
+  - [build]{#gloss-build}\
+    Produce a [store object].
+
+    See [`nix-build`](./command-ref/nix-build.md).
+
+    See [`nix build`](./command-ref/new-cli/nix3-build.md) (experimental).
+
+    Nix uses a two-stage build process:
+    A Nix expression that evaluates to a [derivation] is [instantiate]d to a [store derivation], which is then [realise]d to a [store object].
```

Comment:
What do you think about making this two-stage build more visible with a numbered list?
Incidentally I think it makes it easier to read and grasp the explanation
E.g.

```suggestion
    Nix uses a two-stage build process:
    1. A Nix expression that evaluates to a [derivation] is [instantiate]d to a [store derivation]
    2. This [store derivation] is then [realise]d to a [store object].
```

Reply:
Minor English comment: "realised to" looks weird to me (non-native English speaker), I would have intuitively said "realised into", but I fail to find a convincing source of truth about either